test/support.ts

```diff
@@ -290,19 +290,22 @@ type PartialRecord<K extends keyof any, V> = Partial<Record<K, V>>;
 
 export function expectsql(
   query: { query: string, bind: unknown } | Error,
-  assertions: { query: PartialRecord<ExpectationKey, string | Error>, bind: PartialRecord<ExpectationKey, unknown> },
+  assertions: {
```

Comment:
I'm very sorry, I made changes to our test tools that are going to conflict with this

I don't think the unit test is necessary if you add the integration test, but if you want to do the regex test anyway, I would propose to revert the changes you made to expectsql and instead use the new `expectPerDialect` function & create a new `toMatchRegex` assertion:

```ts
expectPerDialect(firstCall.args[0] as string, {
  default: toMatchRegex(/INSERT INTO (?:`|")Users(?:`|") \((?:`|")firstName(?:`|")\) VALUES (?:\('\w+'\),){999}\('\w+'\);/),
  ibmi: toMatchRegex(/SELECT \* FROM FINAL TABLE \(INSERT INTO "Users" \("firstName"\) VALUES (?:\('\w+'\),){999}\('\w+'\)\)/),
  mssql: toMatchRegex(/INSERT INTO \[Users\] \(\[firstName\]\) VALUES (?:\(N'\w+'\),){999}\(N'\w+'\);/),
})
```

`expectPerDialect` is a more versatile version of `expectsql`

Reply:
Let me know if you are happy with the existing integration test otherwise I will update the regex test